Sensus.UI.Tests/Tools/Tests.cs

```diff
@@ -1,175 +1,178 @@
+        #region Old Tests
+        //[Test, Ignore("Doesn't work. Old Test.")]
+        //public void ReadEvaluatePrintLoop()
+        //{
+        //    _app.Repl();
+        //}
+
+        //[Test, Ignore("Doesn't work. Old Test.")]
+        //public void RunUnitTestingProtocol()
+        //{
+        //    // wait a bit for app to start up -- ios sometimes displays permissions dialogs that need time to be dismissed (UiTest appears to take care of these on its own)
+        //    Thread.Sleep(5000);
+
+        //    // protocol has been started and is waiting for user consent
+        //    ConsentToProtocolStart(new TimeSpan(0, 0, 5));
+
+        //    // stop and edit protocol
+        //    StopProtocol();
+        //    TapProtocol(false);
+        //    _app.WaitForElementThenTap(PROTOCOL_ACTION_SHEET_EDIT, false);
+
+        //    // set data store delays such that they will be testable within a reasonable time period
+        //    TimeSpan localDataStoreDelay = new TimeSpan(0, 0, 5);
+        //    _app.WaitForElementThenTap(LOCAL_DATA_STORE_EDIT, true);
+        //    _app.WaitForElementThenEnterText(DATA_STORE_COMMIT_DELAY, true, localDataStoreDelay.TotalMilliseconds.ToString());
+        //    _app.WaitForElementThenTap(DATA_STORE_OK, true);
+
+        //    TimeSpan remoteDataStoreDelay = new TimeSpan(0, 0, 15);
+        //    _app.WaitForElementThenTap(REMOTE_DATA_STORE_EDIT, true);
+        //    _app.WaitForElementThenEnterText(DATA_STORE_COMMIT_DELAY, true, remoteDataStoreDelay.TotalMilliseconds.ToString());
+        //    _app.WaitForElementThenTap(DATA_STORE_OK, true);  // to protocol page
+        //    _app.Back();  // to protocols page
+
+        //    // restart protocol, wait for remote data store to commit data, and then check status
+        //    StartProtocol(new TimeSpan(0, 0, 5));
+        //    Thread.Sleep(remoteDataStoreDelay.Add(new TimeSpan(0, 0, 5)));
+        //    AssertProtocolStatusEmpty("Protocol status after remote data store.");
+
+        //    // enable accelerometer to check script triggering
+        //    TapProtocol(true);
+        //    _app.WaitForElementThenTap(PROTOCOL_ACTION_SHEET_EDIT, false);
+        //    _app.WaitForElementThenTap(PROBES_EDIT, true);
+        //    _app.WaitForElementThenTap(ACCELEROMETER_PROBE, true);
+        //    _app.WaitForElementThenTap(ACCELEROMETER_ENABLED, true);
+        //    Thread.Sleep(10000);  // 5-second accelerometer stabilization + a few seconds for the prompt page to show up before trying to scroll to it.
+        //    _app.WaitForElementThenEnterText(ACCELEROMETER_TRIGGERED_SCRIPT_INPUT, true, "12345");
+        //    _app.WaitForElementThenTap(PROMPT_FOR_INPUTS_SUBMIT, true);
+        //    _app.WaitForElementThenTap(PROMPT_FOR_INPUTS_SUBMIT_CONFIRM, false);
+        //    _app.Back();  // to probes page
+        //    _app.Back();  // to protocol page
+        //    _app.Back();  // to protocols page
+
+        //    StopProtocol();
+        //}

```

Comment:
Yeah, I thought the same thing. I spent a few hours trying to figure it out and never got it to work.  Below is what I tried and what I mean by Integration vs Unit testing. There may be a way but I was never able to figure it out. I'd love to hear any ideas you might have.

I was able to get their UITest framework working, but as I said it was a complete integration test.  Unfortunately, the terminology gets a little flimsy since marketers like to use the word "unit test." In my professional experience Unit Testing is what you call exercising code in small pieces to make sure each component is functional. "Integration Testing" is typically used to refer to testing that treats a piece of software as a whole unit. Individual code pieces aren't tested. Integration testing tends to be more brittle because the interactions between components can change without changing an application's functionality. For Example, the UI layout can change without us changing the code or we could replace our back end remote data store without changing the code. Each of these changes could likely break a Xamarin UITest while not likely not having any effect on component level testing.

What I wanted was a way to inject and run small pieces of code on the emulators. The only way that XTC is able to test is by installing the whole application and then providing the IApp interface which gives developers programatic control with their app. It does use the NUnit framework to create Assert statements, but I was never able to find a way to test specific lines of code.